subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/InputFilePropertySpec.java

```diff
@@ -16,6 +16,10 @@
 
 package org.gradle.api.internal.tasks.properties;
 
+import javax.annotation.Nullable;
+
 public interface InputFilePropertySpec extends FilePropertySpec {
     boolean isSkipWhenEmpty();
+    @Nullable
+    Object getValue();
```

Comment:
This doesn't sound like a good idea. `PropertySpec` is about the definition of the property (hence `Spec`), not its value. I see that we are doing something different already with `InputPropertySpec`, but... it just doesn't feel right.

Can we use a `Map` or something instead?

Reply:
Currently, property spec captures the definition of a property. The definition includes the `PropertyValue` and the metadata (name, normalizer, etc.) of the property. For file properties, we only have been capturing the value as a file collection. Would it be better for you if it was possible to query the `PropertyValue` instead of the actual value?